shared struct field of array type with default value causes C error

### Describe the bug

A shared struct field of array type with a default value causes a C error.

### Expected Behavior

Successful compilation

### Current Behavior

```sh
/tmp/v_501/i1.1604850369844756738.tmp.c:17577:46: error: initializing '__shared__Array_int *' (aka 'struct __shared__Array_int *') with an expression of incompatible type 'array' (aka 'struct array')
        {main__StructA _ = ((main__StructA){.ints = new_array_from_c_array_noscan(1, 1, sizeof(int), _MOV((int[1]){0})),});}
                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

### Reproduction Steps

```v
module main

struct StructA{
	ints shared []int = [0]
}

fn main() {
	_ = StructA{}
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

27e1c20

### Environment details (OS name and version, etc.)

V full version: V 0.3.3 2597efa.27e1c20
OS: macos, macOS, 13.3.1, 22E261
CC version: Apple clang version 14.0.3 (clang-1403.0.22.14.1)
thirdparty/tcc status: thirdparty-macos-arm64 a668e5a0

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -161,9 +161,7 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 			g.is_shared = old_is_shared2
 		}
 		for mut field in info.fields {
-			if !field.typ.has_flag(.shared_f) {
-				g.is_shared = false
-			}
+			g.is_shared = field.typ.has_flag(.shared_f)
 			if mut sym.info is ast.Struct {
 				mut found_equal_fields := 0
 				for mut sifield in sym.info.fields {

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -278,7 +278,6 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 				g.write(',')
 			}
 			initialized = true
-			g.is_shared = old_is_shared
 		}
 		g.is_shared = old_is_shared
 	}

--- a/vlib/v/tests/shared_struct_field_test.v
+++ b/vlib/v/tests/shared_struct_field_test.v
@@ -1,0 +1,13 @@
+module main
+
+struct StructA {
+	ints shared []int = [0]
+}
+
+fn test_main() {
+	a := StructA{}
+	rlock a.ints {
+		dump(a)
+		assert true
+	}
+}

